util/version/version.go

```diff
@@ -173,17 +173,57 @@ func (v buildIdentifier) compare(o buildIdentifier) int {
 	}
 }
 
-// CompareWithBuildIdentifiers compares 2 version a and b.
+// compareWithBuildIdentifiers compares 2 version a and b.
 // Perfoms a standard version compare between a and b. If the versions
 // are equal, build identifiers will be used to compare further.
 //   -1 == a is less than b.
 //   0 == a is equal to b.
 //   1 == a is greater than b.
-func CompareWithBuildIdentifiers(a semver.Version, b semver.Version) int {
+func compareWithBuildIdentifiers(a semver.Version, b semver.Version) int {
```

Comment:
Just noticed that we are making this func private without deprecation.
I think we should deprecate this before making it private